atlp-rwanda/atlp-pulse-bn
# #326 fx-remove duplicate codes by using code helper

PR Description
This pull request aims to reduce code duplication by centralizing common functionalities in the codehelper.ts file. Specifically, we've moved the email validation regex and token generation function to codehelper.ts and updated userresolver.ts to use these imported functions.
Description of tasks that were expected to be completed

 Move emailExpression regex to codehelper.ts
 Export emailExpression from codehelper.ts
 Update userresolver.ts to import and use emailExpression
 Update token generation in userresolver.ts to use imported generateToken function

Functionality
The codehelper.ts file now exports two main functionalities:

emailExpression: A regex for validating email addresses
generateToken: A function for generating JWT tokens

The userresolver.ts file now imports these functionalities and uses them instead of duplicating the code. This centralizes these common operations, making the codebase more maintainable and consistent.
How has this been tested?
To test these changes:

Run unit tests for codehelper.ts to ensure emailExpression and generateToken work as expected.
Test the user registration flow in userresolver.ts:

Attempt to register with valid and invalid email addresses
Verify that tokens are generated correctly for new users
Ensure that the generated tokens can be validated and contain the correct user information



PR Checklist:

 Update codehelper.ts with new exports
 Modify userresolver.ts to use imported functionalities
 Run linter to ensure code style consistency
 Update any affected unit tests
 Perform manual testing of the user registration flow

Track PR
Trello Link (#DP-326)
Screenshots (If appropriate)
N/A for this code refactoring PR
